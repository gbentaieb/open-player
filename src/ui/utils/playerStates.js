import * as states from '../constants/CoreStates';

export const playingStates = [states.PLAYING, states.BUFFERING, states.SEEKING];
export const waitingStates = [states.LOADING, states.BUFFERING, states.SEEKING];

export function isWaiting(state) {
  return waitingStates.includes(state);
}

export function isPlaying(state) {
  return playingStates.includes(state);
}
