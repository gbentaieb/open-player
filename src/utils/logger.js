/* eslint-disable no-console */

class Logger {
  defaultLevel = 5;

  LEVELS = {
    LOG: 0,
    DEBUG: 1,
    INFO: 2,
    WARN: 3,
    ERROR: 4,
    OFF: 5,
  };

  constructor() {
    this.currentLevel = this.defaultLevel;
  }

  static get instance() {
    if (!this.singleton) {
      this.singleton = new Logger();
    }

    return this.singleton;
  }

  setLevel(value) {
    if (typeof value === 'string') {
      this.currentLevel = this.LEVELS[value];
    } else {
      this.currentLevel = value;
    }
  }

  isEnabled(level) {
    return level >= this.currentLevel;
  }

  log(msg, ...args) {
    const { LOG } = this.LEVELS;

    if (this.isEnabled(LOG)) {
      console.log(msg, ...args);
    }
  }

  debug(msg, ...args) {
    const { DEBUG } = this.LEVELS;

    if (this.isEnabled(DEBUG)) {
      console.debug(msg, ...args);
    }
  }

  info(msg, ...args) {
    const { INFO } = this.LEVELS;

    if (this.isEnabled(INFO)) {
      console.info(msg, ...args);
    }
  }

  warn(msg, ...args) {
    const { WARN } = this.LEVELS;

    if (this.isEnabled(WARN)) {
      console.warn(msg, ...args);
    }
  }

  error(msg, ...args) {
    const { ERROR } = this.LEVELS;

    if (this.isEnabled(ERROR)) {
      console.error(msg, ...args);
    }
  }
}

const logger = Logger.instance;

export default logger;
