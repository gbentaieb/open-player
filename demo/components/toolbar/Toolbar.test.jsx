import React from 'react';
import { spy } from 'sinon';
import TextField from 'material-ui/TextField/TextField';
import Toolbar from './Toolbar';
import presets from '../../presets';
import { MuiMountWithContext } from '../../../test/utils';

test('Toolbar > Snapshot', () => {
  const wrapper = shallow(<Toolbar loadVideo={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('Toolbar > handleChange', () => {
  const wrapper = shallow(<Toolbar loadVideo={() => {}} />);
  expect(wrapper.state('value')).toEqual(presets[1].value);
  wrapper.instance().handleChange('', '', 'test');
  expect(wrapper.state('value')).toEqual('test');
});

test('Toolbar > text field disabled', () => {
  const wrapper = MuiMountWithContext(<Toolbar loadVideo={() => {}} />);
  const textField = wrapper.find(TextField);
  expect(textField.length).toEqual(0);
});

test('Toolbar > text field enabled', () => {
  const wrapper = MuiMountWithContext(<Toolbar loadVideo={() => {}} />);
  wrapper.setState({ value: null });
  const textField = wrapper.find(TextField);
  expect(textField.length).toEqual(1);
});

test('Toolbar > onLoadClicked with value', () => {
  const loadVideo = spy();
  const value = 'www.test.com/test.mp4';
  const wrapper = MuiMountWithContext(<Toolbar loadVideo={loadVideo} />);
  wrapper.setState({ value });
  wrapper.instance().onLoadClicked();
  expect(loadVideo.calledWith(value)).toEqual(true);
});

test('Toolbar > onLoadClicked without value', () => {
  const loadVideo = spy();
  const value = 'www.test.com/test.mp4';
  const wrapper = MuiMountWithContext(<Toolbar loadVideo={loadVideo} />);
  wrapper.setState({ value: null });
  wrapper.instance().TextField.value = value;
  wrapper.instance().onLoadClicked();
  expect(loadVideo.calledWith(value)).toEqual(true);
});
