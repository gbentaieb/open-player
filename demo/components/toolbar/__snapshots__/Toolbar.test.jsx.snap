// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test demo toolbar Toolbar > Snapshot 1`] = `
<div
  className="Toolbar"
>
  <Toolbar
    noGutter={false}
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <ToolbarGroup
      firstChild={true}
      lastChild={false}
      style={
        Object {
          "margin": 0,
          "width": "100%",
        }
      }
    >
      <DropDownMenu
        anchorOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        animated={true}
        autoWidth={true}
        disabled={false}
        iconButton={<NavigationArrowDropDown />}
        maxHeight={500}
        multiple={false}
        onChange={[Function]}
        openImmediately={false}
        style={
          Object {
            "margin": 0,
            "overflow": "hidden",
          }
        }
        value="https://dash.akamaized.net/dash264/TestCases/2c/qualcomm/2/MultiRes.mpd"
      >
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          key="0"
          primaryText="Custom Url"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value={null}
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          key="1"
          primaryText="DASH - Big Buck Bunny"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="https://dash.akamaized.net/dash264/TestCases/2c/qualcomm/2/MultiRes.mpd"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          key="2"
          primaryText="DASH - The first Blender Open Movie from 2006"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="https://dash.akamaized.net/dash264/TestCases/2c/qualcomm/1/MultiResMPEG2.mpd"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          key="3"
          primaryText="DASH - Test"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="https://dash.akamaized.net/dash264/CTA/imsc1/IT1-20171027_dash.mpd"
        />
      </DropDownMenu>
      <ToolbarSeparator />
      <RaisedButton
        disabled={false}
        fullWidth={false}
        label="Load"
        labelPosition="after"
        onClick={[Function]}
        primary={true}
        secondary={false}
      />
    </ToolbarGroup>
  </Toolbar>
  <br />
</div>
`;
